client/blocks/edit-gravatar/index.jsx

```diff
@@ -156,40 +158,82 @@ export class EditGravatar extends Component {
 		}
 	}
 
+	renderEmailVerificationNotice = () => {
+		const user = {
+			data: this.props.user
+		};
+		return (
+			<VerifyEmailDialog
+				user={ user }
+				onClose={ this.closeVerifyEmailDialog }
+			/>
+		);
+	}
+
+	handleClick = () => {
+		if ( this.props.user.email_verified ) {
+			return;
+		}
+		this.setState( () => ( {
+			showEmailVerificationNotice: true
+		} ) );
+	}
+
+	closeVerifyEmailDialog = () => {
+		this.setState( () => ( {
+			showEmailVerificationNotice: false
+		} ) );
+	}
+
 	render() {
 		const {
 			isUploading,
 			translate,
 			user
 		} = this.props;
+
 		const gravatarLink = `//gravatar.com/${ user.username || '' }`;
+		const icon = ( user.email_verified ? 'cloud-upload' : 'notice' );
+		const buttonText = ( user.email_verified
+			? translate( 'Click to change photo' )
+			: translate( 'Verify email first' )
+		);
+
 		return (
-			<div className="edit-gravatar">
-				{ this.renderImageEditor() }
-				<FilePicker accept="image/*" onPick={ this.onReceiveFile }>
-					<div
-						className={
-							classnames( 'edit-gravatar__image-container',
-								{ 'is-uploading': isUploading }
-							)
-						}
-					>
-						<Gravatar
-							imgSize={ 400 }
-							size={ 150 }
-							user={ user }
-						/>
-						{ ! isUploading && (
-							<div className="edit-gravatar__label-container">
-								<Gridicon icon="cloud-upload" size={ 36 } />
-								<span className="edit-gravatar__label">
-									{ this.props.translate( 'Click to change photo' ) }
-								</span>
+			<div
+				className={
+					classnames( 'edit-gravatar',
+						{ 'is-uploading': isUploading },
+						{ 'edit-gravatar__unverified-user': ! user.email_verified }
```

Comment:
Since it's not creating any new element `is-unverified` sounds better. And it's fine to do:

```sass
.edit-gravatar.is-unverified {
    .file-picker {}
    .gravatar {}
}
```